nlu.py

```diff
@@ -20,6 +20,38 @@ def _get_commands(text_sequence: [str]) -> set:
     return possible_commands
 
 
+def _consine_similarity(a: [str], b: [str]) -> float:
+    words = set(a+b)
+    vector_a = _word2vector(words, a)
+    vector_b = _word2vector(words, b)
+    return _cosine_theta(vector_a, vector_b)
+
+
+def _cosine_theta(a: [int], b: [int]):
+    numerator = sum(x1 * x2 for x1, x2 in zip(a, b))
+    denominator = _dot(a)*_dot(b)
+    return numerator/denominator
+
+
+def _dot(v: [int]) -> float:
+    return sum(pow(n, 2) for n in v) ** 0.5
+
+
+def _word2vector(all_words: set, text_sequence: [str]) -> [int]:
+    frequency = dict.fromkeys(all_words, 0)
+    for text in text_sequence:
+        if text in all_words:
+            value = frequency.get(text)
+            if value is None:
+                value = 1
+            else:
+                value += 1
+            frequency[text] = value
```

Comment:
```py
frquency[text] = frequency.get(text, 0) + 1
```
Then you can replace 44 to 49 by this.

Reply:
How about
```py
a = ['a', 'b', 'c']
b = ['a', 'b', 'd']

all_words = set(a+b)


def get_vec(all_words: set, seq: [str]) -> [int]:
    vec = [0] * len(all_words)
    for i, word in enumerate(all_words):
        for v in seq:
            if v == word:
                vec[i] += 1
    return vec


print(get_vec(all_words, a))
print(get_vec(all_words, b))
```

Reply:
It's amazing. It's more simple and performance.